paddle/go/optimizer/optimizer_factory.cc

```diff
@@ -0,0 +1,66 @@
+#include "optimizer_factory.h"
+
+namespace paddle {
+namespace optimizer {
+
+template<class T>
+SGDOptimizer<T>::SGDOptimizer(const ::paddle::OptimizerConfig &config) : ParameterOptimizer<T>(config) {
+  learning_rate = config.learning_rate();
+  momentum = config.momentum();
+  decay = config.decay();
+  nesterov = config.nesterov();
+  lr_decay_a = config.lr_decay_a();
+  lr_decay_b = config.lr_decay_b();
+}
+
+template<class T>
+void SGDOptimizer<T>::destroy() {
```

Comment:
thanks pointed out that! You are right, we do not need a destroy explicitly. 
just call the destructor with `delete xxx` is more elegant.  I made a mistake here, I thought that In C interface we need an explicitly destroy function call to pass the GCC compile, in fact, I write the cmake script in wrong way. fix it.
Thanks for you reminding!